pkg/reconciler/v1alpha1/service/service.go

```diff
@@ -207,7 +207,16 @@ func (c *Reconciler) reconcile(ctx context.Context, service *v1alpha1.Service) e
 	}
 
 	// Update our Status based on the state of our underlying Route.
-	service.Status.PropagateRouteStatus(route.Status)
+	switch {
+	case service.Spec.RunLatest != nil:
+		// In case of RunLatest, verify that traffic has already migrated
+		// to the latest revision.
+		service.Status.PropagateRouteStatus(&route.Status, func() bool {
+			return len(route.Status.Traffic) > 0 && route.Status.Traffic[0].RevisionName == config.Status.LatestReadyRevisionName
+		})
+	default:
+		service.Status.PropagateRouteStatus(&route.Status, nil /*no Route verification callback*/)
+	}
```

Comment:
Should the switch just define a `var routeVerificationFunc` vs copying calls to `PropagateRouteStatus`?